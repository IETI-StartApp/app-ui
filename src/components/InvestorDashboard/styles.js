import { makeStyles } from '@material-ui/core';

export const useStyles = makeStyles(() => ({
    chart:{
        paddingRight: '20px',
    }
}));