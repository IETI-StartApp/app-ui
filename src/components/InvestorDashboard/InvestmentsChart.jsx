import React from 'react';
import {
    Chart,
    ArgumentAxis,
    ValueAxis,
    LineSeries,
} from '@devexpress/dx-react-chart-material-ui';
import { Paper } from '@material-ui/core';
import { useStyles } from './styles';
import { Animation } from '@devexpress/dx-react-chart';

export const InvestmentsChart = () => {
    const classes = useStyles();
    const data = [
        {
            Month: "Diciembre", Total: 0,
        }, {
            Month: "Enero", Total: 0,
        }, {
            Month: "Febrero", Total: 0,
        }, {
            Month: "Marzo", Total: 700000,
        }, {
            Month: "Abril", Total: 1800000,
        }, {
            Month: "Mayo", Total: 1000000,
        },
    ];

    return (
        <>
            <Paper>
                <Chart
                    data={data}
                    className={classes.chart}
                >
                    <ArgumentAxis />
                    <ValueAxis />
                    <LineSeries valueField="Total" argumentField="Month" color="#950740"/>
                    <Animation />
                </Chart>
            </Paper>
        </>
    );
}