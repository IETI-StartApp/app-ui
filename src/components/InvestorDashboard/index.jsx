import React from 'react';
import Navigation from '../Navbar/Navigation';
import { useStyles } from './styles';
import { Card, CardContent, Typography, Avatar, Grid, CardActionArea, Box, GridList, GridListTile, Fab } from '@material-ui/core/';
import AttachMoneyIcon from '@material-ui/icons/AttachMoney';
import StarBorderIcon from '@material-ui/icons/StarBorder';
import { useHistory } from "react-router";
import TrendingUpIcon from '@material-ui/icons/TrendingUp';
import BookmarkBorderIcon from '@material-ui/icons/BookmarkBorder';
import { InvestmentsChart } from "./InvestmentsChart";
import CardProject from '../ProjectItems/CardProject';
import ChatBubbleOutlineIcon from '@material-ui/icons/ChatBubbleOutline';

export const InvestorDashboard = () => {
    const history = useHistory() || [];
    const classes = useStyles();
    const items = [
        { number: "4", description: "Proyectos Financiados", icon: <StarBorderIcon fontSize="large" />, path: "/" },
        { number: "3.5M", description: "Capital invertido", icon: <AttachMoneyIcon fontSize="large" />, path: "/dashboard" },
        { number: "1000", description: "Oportunidades de inversión ", icon: <TrendingUpIcon fontSize="large" />, path: "/" },
        { number: "2", description: "Proyectos guardados", icon: <BookmarkBorderIcon fontSize="large" />, path: "/" }
    ];
    const projects = [
        {
            name: "Proyecto 1", 
            image:"https://www.binarias.org/wp-content/uploads/2020/12/tegnologia.jpeg", 
            finance: { value:15000000, valuation:20000000, startDate: "10/03/2021"},
        },
        {
            name: "Proyecto 2", 
            image:"https://s3-us-west-2.amazonaws.com/wp-mpro-blog/wp-content/uploads/2016/03/14133959/s3-blog-tecnologia-como-prioridad-min.png", 
            finance: { value:10000000, valuation:17500000, startDate: "12/02/2021"},
        },
        {
            name: "Proyecto 3", 
            image:"https://thelogisticsworld.com/wp-content/uploads/2020/02/empresario-analizando-informe-financiero-empresa-graficos-realidad-aumentada_34141-360.jpg", 
            finance: { value:8000000, valuation:12000000, startDate: "21/01/2021"},
        },
    ];

    return (
        <>
            <Navigation />
            <div className={classes.offset}></div>
            <div className={classes.root}>
                <Grid container spacing={2} direction="col" justify="center" alignItems="center" className={classes.allItems}>
                    <Grid item xs={12}>
                        <Grid container spacing={2} direction="row" justify="center" className={classes.items}>
                            {
                                items.map(item => (
                                    <Grid item xs={12} sm={5} lg={3}>
                                        <div style={{ display: "flex", alignItems: "center", justifyContent: "center" }}>
                                            <Card variant="outlined" className={classes.dashItem}>
                                                <Avatar className={classes.cardAvatar} variant="rounded">
                                                    {item.icon}
                                                </Avatar>
                                                <CardActionArea onClick={() => { history.push(item.path) }}>
                                                    <CardContent className={classes.itemContent}>
                                                        <Typography variant="h4" align="center" className={classes.mainColor} >{item.number}</Typography>
                                                        <Typography align="center">{item.description}</Typography>
                                                    </CardContent>
                                                </CardActionArea>
                                            </Card>
                                        </div>
                                    </Grid>
                                ))
                            }
                        </Grid>
                    </Grid>
                    <Grid item xs={12}>
                        <Grid container spacing={2} direction="row" justify="center">
                            <Grid item xs={12} sm={10} lg={6}>
                                <Card variant="outlined" className={classes.cardChart}>
                                    <CardContent>
                                        <Typography className={classes.mainColor} variant="button" display="block">Inversiones en los últimos 6 meses</Typography>
                                        <Box p={3}>
                                            <InvestmentsChart />
                                        </Box>
                                    </CardContent>
                                </Card>
                            </Grid>
                            <Grid item xs={12} sm={10} lg={6}>
                                <Card variant="outlined" className={classes.projects}>
                                    <CardContent>
                                        <Typography className={classes.mainColor} variant="button" display="block">Oportunidades de inversión</Typography>
                                            <div className={classes.rootProjects}>
                                                <GridList  className={classes.gridList}>
                                                    {projects.map((project) => (
                                                        <GridListTile className={classes.gridListTile}>
                                                            <CardProject project={project} />
                                                        </GridListTile>
                                                    ))}
                                                </GridList>
                                            </div>
                                    </CardContent>
                                </Card>
                            </Grid>
                        </Grid>
                    </Grid>
                </Grid>
                <Fab aria-label="Chat" className={classes.fab}>
                    <ChatBubbleOutlineIcon />
                </Fab>
            </div>
        </>
    );
}