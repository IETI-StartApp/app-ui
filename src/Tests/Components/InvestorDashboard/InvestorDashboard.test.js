import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import { InvestorDashboard } from '../../../components/InvestorDashboard/index';

describe('Testing navigation', () => {

    const wrapper = shallow(<InvestorDashboard />)

    test('should load navigation ', () => {
        expect(wrapper).toMatchSnapshot();
    })
})
