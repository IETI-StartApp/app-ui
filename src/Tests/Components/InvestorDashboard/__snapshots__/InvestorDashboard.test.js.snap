// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing navigation should load navigation  1`] = `
<Fragment>
  <Navigation />
  <div
    className="makeStyles-offset-14"
  />
  <div
    className="makeStyles-root-1"
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      className="makeStyles-allItems-8"
      container={true}
      direction="col"
      justify="center"
      spacing={2}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <WithStyles(ForwardRef(Grid))
          container={true}
          direction="row"
          justify="center"
          spacing={2}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            lg={3}
            sm={5}
            xs={12}
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "center",
                }
              }
            >
              <WithStyles(ForwardRef(Card))
                className="makeStyles-dashItem-2"
                variant="outlined"
              >
                <WithStyles(ForwardRef(Avatar))
                  className="makeStyles-cardAvatar-4"
                  variant="rounded"
                >
                  <Memo([object Object])
                    fontSize="large"
                  />
                </WithStyles(ForwardRef(Avatar))>
                <WithStyles(ForwardRef(CardActionArea))
                  onClick={[Function]}
                >
                  <WithStyles(ForwardRef(CardContent))
                    className="makeStyles-itemContent-3"
                  >
                    <WithStyles(ForwardRef(Typography))
                      align="center"
                      className="makeStyles-mainColor-7"
                      variant="h4"
                    >
                      4
                    </WithStyles(ForwardRef(Typography))>
                    <WithStyles(ForwardRef(Typography))
                      align="center"
                    >
                      Proyectos Financiados
                    </WithStyles(ForwardRef(Typography))>
                  </WithStyles(ForwardRef(CardContent))>
                </WithStyles(ForwardRef(CardActionArea))>
              </WithStyles(ForwardRef(Card))>
            </div>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            lg={3}
            sm={5}
            xs={12}
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "center",
                }
              }
            >
              <WithStyles(ForwardRef(Card))
                className="makeStyles-dashItem-2"
                variant="outlined"
              >
                <WithStyles(ForwardRef(Avatar))
                  className="makeStyles-cardAvatar-4"
                  variant="rounded"
                >
                  <Memo([object Object])
                    fontSize="large"
                  />
                </WithStyles(ForwardRef(Avatar))>
                <WithStyles(ForwardRef(CardActionArea))
                  onClick={[Function]}
                >
                  <WithStyles(ForwardRef(CardContent))
                    className="makeStyles-itemContent-3"
                  >
                    <WithStyles(ForwardRef(Typography))
                      align="center"
                      className="makeStyles-mainColor-7"
                      variant="h4"
                    >
                      3.5M
                    </WithStyles(ForwardRef(Typography))>
                    <WithStyles(ForwardRef(Typography))
                      align="center"
                    >
                      Capital invertido
                    </WithStyles(ForwardRef(Typography))>
                  </WithStyles(ForwardRef(CardContent))>
                </WithStyles(ForwardRef(CardActionArea))>
              </WithStyles(ForwardRef(Card))>
            </div>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            lg={3}
            sm={5}
            xs={12}
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "center",
                }
              }
            >
              <WithStyles(ForwardRef(Card))
                className="makeStyles-dashItem-2"
                variant="outlined"
              >
                <WithStyles(ForwardRef(Avatar))
                  className="makeStyles-cardAvatar-4"
                  variant="rounded"
                >
                  <Memo([object Object])
                    fontSize="large"
                  />
                </WithStyles(ForwardRef(Avatar))>
                <WithStyles(ForwardRef(CardActionArea))
                  onClick={[Function]}
                >
                  <WithStyles(ForwardRef(CardContent))
                    className="makeStyles-itemContent-3"
                  >
                    <WithStyles(ForwardRef(Typography))
                      align="center"
                      className="makeStyles-mainColor-7"
                      variant="h4"
                    >
                      1000
                    </WithStyles(ForwardRef(Typography))>
                    <WithStyles(ForwardRef(Typography))
                      align="center"
                    >
                      Oportunidades de inversión 
                    </WithStyles(ForwardRef(Typography))>
                  </WithStyles(ForwardRef(CardContent))>
                </WithStyles(ForwardRef(CardActionArea))>
              </WithStyles(ForwardRef(Card))>
            </div>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            lg={3}
            sm={5}
            xs={12}
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "center",
                }
              }
            >
              <WithStyles(ForwardRef(Card))
                className="makeStyles-dashItem-2"
                variant="outlined"
              >
                <WithStyles(ForwardRef(Avatar))
                  className="makeStyles-cardAvatar-4"
                  variant="rounded"
                >
                  <Memo([object Object])
                    fontSize="large"
                  />
                </WithStyles(ForwardRef(Avatar))>
                <WithStyles(ForwardRef(CardActionArea))
                  onClick={[Function]}
                >
                  <WithStyles(ForwardRef(CardContent))
                    className="makeStyles-itemContent-3"
                  >
                    <WithStyles(ForwardRef(Typography))
                      align="center"
                      className="makeStyles-mainColor-7"
                      variant="h4"
                    >
                      2
                    </WithStyles(ForwardRef(Typography))>
                    <WithStyles(ForwardRef(Typography))
                      align="center"
                    >
                      Proyectos guardados
                    </WithStyles(ForwardRef(Typography))>
                  </WithStyles(ForwardRef(CardContent))>
                </WithStyles(ForwardRef(CardActionArea))>
              </WithStyles(ForwardRef(Card))>
            </div>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <WithStyles(ForwardRef(Grid))
          container={true}
          direction="row"
          justify="center"
          spacing={2}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            lg={6}
            sm={10}
            xs={12}
          >
            <WithStyles(ForwardRef(Card))
              className="makeStyles-cardChart-5"
              variant="outlined"
            >
              <WithStyles(ForwardRef(CardContent))>
                <WithStyles(ForwardRef(Typography))
                  className="makeStyles-mainColor-7"
                  display="block"
                  variant="button"
                >
                  Inversiones en los últimos 6 meses
                </WithStyles(ForwardRef(Typography))>
                <Styled(MuiBox)
                  p={3}
                >
                  <InvestmentsChart />
                </Styled(MuiBox)>
              </WithStyles(ForwardRef(CardContent))>
            </WithStyles(ForwardRef(Card))>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            lg={6}
            sm={10}
            xs={12}
          >
            <WithStyles(ForwardRef(Card))
              className="makeStyles-projects-6"
              variant="outlined"
            >
              <WithStyles(ForwardRef(CardContent))>
                <WithStyles(ForwardRef(Typography))
                  className="makeStyles-mainColor-7"
                  display="block"
                  variant="button"
                >
                  Oportunidades de inversión
                </WithStyles(ForwardRef(Typography))>
                <div
                  className="makeStyles-rootProjects-10"
                >
                  <WithStyles(ForwardRef(GridList))
                    className="makeStyles-gridList-11"
                  >
                    <WithStyles(ForwardRef(GridListTile))
                      className="makeStyles-gridListTile-12"
                    >
                      <CardProject
                        project={
                          Object {
                            "finance": Object {
                              "startDate": "10/03/2021",
                              "valuation": 20000000,
                              "value": 15000000,
                            },
                            "image": "https://www.binarias.org/wp-content/uploads/2020/12/tegnologia.jpeg",
                            "name": "Proyecto 1",
                          }
                        }
                      />
                    </WithStyles(ForwardRef(GridListTile))>
                    <WithStyles(ForwardRef(GridListTile))
                      className="makeStyles-gridListTile-12"
                    >
                      <CardProject
                        project={
                          Object {
                            "finance": Object {
                              "startDate": "12/02/2021",
                              "valuation": 17500000,
                              "value": 10000000,
                            },
                            "image": "https://s3-us-west-2.amazonaws.com/wp-mpro-blog/wp-content/uploads/2016/03/14133959/s3-blog-tecnologia-como-prioridad-min.png",
                            "name": "Proyecto 2",
                          }
                        }
                      />
                    </WithStyles(ForwardRef(GridListTile))>
                    <WithStyles(ForwardRef(GridListTile))
                      className="makeStyles-gridListTile-12"
                    >
                      <CardProject
                        project={
                          Object {
                            "finance": Object {
                              "startDate": "21/01/2021",
                              "valuation": 12000000,
                              "value": 8000000,
                            },
                            "image": "https://thelogisticsworld.com/wp-content/uploads/2020/02/empresario-analizando-informe-financiero-empresa-graficos-realidad-aumentada_34141-360.jpg",
                            "name": "Proyecto 3",
                          }
                        }
                      />
                    </WithStyles(ForwardRef(GridListTile))>
                  </WithStyles(ForwardRef(GridList))>
                </div>
              </WithStyles(ForwardRef(CardContent))>
            </WithStyles(ForwardRef(Card))>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Fab))
      aria-label="Chat"
      className="makeStyles-fab-13"
    >
      <Memo([object Object]) />
    </WithStyles(ForwardRef(Fab))>
  </div>
</Fragment>
`;
